sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -0,0 +1,377 @@
+    return key !== CoreHttpConstants.XML_METADATA_MARKER;
+  });
+
+  if (contentElementNames && contentElementNames[0]) {
+    const contentRootElementName = contentElementNames[0];
+    delete entry.content[contentRootElementName][CoreHttpConstants.XML_METADATA_MARKER];
+    result = entry.content[contentRootElementName];
+
+    if (result) {
+      if (entry[CoreHttpConstants.XML_METADATA_MARKER]) {
+        result[ServiceBusConstants.ATOM_METADATA_MARKER] =
+          entry[CoreHttpConstants.XML_METADATA_MARKER];
+      } else {
+        result[ServiceBusConstants.ATOM_METADATA_MARKER] = {};
+      }
+
+      result[ServiceBusConstants.ATOM_METADATA_MARKER][
+        "ContentRootElement"
+      ] = contentRootElementName;
+
+      Object.keys(entry).forEach((property: string) => {
+        if (property !== "content" && property !== CoreHttpConstants.XML_METADATA_MARKER) {
+          result[ServiceBusConstants.ATOM_METADATA_MARKER][property] = entry[property];
+        }
+      });
+
+      return result;
+    }
+  }
+
+  return undefined;
+}
+
+/**
+ * @ignore
+ * Utility to help parse given `feed` result
+ * @param feed
+ */
+function parseFeedResult(feed: any): object[] {
+  const result = [];
+  if (typeof feed === "object" && feed != null && feed.entry) {
+    if (Array.isArray(feed.entry)) {
+      feed.entry.forEach((entry: any) => {
+        const parsedEntryResult = parseEntryResult(entry);
+        if (parsedEntryResult) {
+          result.push(parsedEntryResult);
+        }
+      });
+    } else {
+      const parsedEntryResult = parseEntryResult(feed.entry);
+      if (parsedEntryResult) {
+        result.push(parsedEntryResult);
+      }
+    }
+  }
+  return result;
+}
+
+function isKnownResponseCode(
+  statusCode: number
+): statusCode is keyof typeof ServiceBusConstants.HttpResponseCodes {
+  return !!(ServiceBusConstants.HttpResponseCodes as { [statusCode: number]: string })[statusCode];
+}
+
+/**
+ * @ignore
+ * Extracts the applicable entity name(s) from the URL based on the known structure
+ * and instantiates the corresponding name properties to the deserialized response
+ *
+ * For instance, following is the URL structure for when creating a rule
+ * `<namespace-component>/<topic-name>/Subscriptions/<subscription-name>/Rules/<rule-name>`
+ *
+ * @param entry
+ * @param nameProperties
+ */
+function setName(entry: any, nameProperties: any): any {
+  if (entry[ServiceBusConstants.ATOM_METADATA_MARKER]) {
+    const parsedUrl = new URL(entry[ServiceBusConstants.ATOM_METADATA_MARKER].id);
+
+    let pathname: string = parsedUrl.pathname;
+
+    // The parsedUrl gets constructed differently for browser vs Node.
+    // It is specifically behaves different for some of the Atom based management API where
+    // the received URL in "id" element is of type "sb:// ... " and not a standard HTTP one
+    if (!isNode) {
+      if (pathname.startsWith("//")) {
+        pathname = pathname.substring(3);
```

Comment:
Say the id property had the value 

`sb://namespace/topicname/Subscriptions/subscriptioname?api-version=2017-04`

From my test runs, in browser `parsedUrl.pathname` would be `//namespace/topicname/Subscriptions/subscriptioname`

Doing a substring(3) on this would give you `amespace/topicname/Subscriptions/subscriptioname`

But, what we want is `topicname/Subscriptions/subscriptioname` right?

A safer approach would be to replace the `sb://` with say `https://` before parsing the url. Then both node & browser should be able to use the same code

Reply:
No we want the string preceding the  `/` before 'topicName` as well, the substring is to specifically remove the `//` left over due to `sb://` being present as the protocol.
I think the service has been peculiar in how it constructs this for different ops as well iirc.
Agree with updating to replace the unexpected `sb://`